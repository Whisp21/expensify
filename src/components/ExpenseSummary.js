import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import numeral from "numeral";
import selectExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expenses-total";

export const ExpenseSummary = ({ expensesCount, expensesTotal }) => {
  const expensesWord = expensesCount === 1 ? "expense" : "expenses";
  const expensesTotalFormat = numeral(expensesTotal / 100).format("0,0.00")

  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title"><span>{expensesCount}</span> {expensesWord} totalling <span>R {expensesTotalFormat}</span></h1>
          <div className="page-header__actions">
            <Link to="/create" className="button add--button">Add Expense</Link>
          </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expensesCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpenseSummary);
