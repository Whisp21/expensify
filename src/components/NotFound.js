import React from "react";
import { Link } from "react-router-dom";

const NotFound = () => (
  <div>
  Error 404 - Not Found
  <Link to="/">Home Page</Link>
  </div>
);

export default NotFound;
