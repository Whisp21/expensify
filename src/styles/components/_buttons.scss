.button {
  background-color: $blue;
  border: none;
  border-radius: 3px;
  color: white;
  font-weight: 300;
  font-size: $font-size-large;
  padding: $s-size;
  width: 18rem;
}

.add--button {
  line-height: 1;
  text-decoration: none;
}

.button--link {
  background: none;
}

.google--button {
  background-color: $google;
}

.button--secondary {
  background-color: #888;
}
