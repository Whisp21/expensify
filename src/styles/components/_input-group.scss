.input-group {
  display: flex;
  margin-bottom: $m-size;
  flex-direction: column;
  @media (min-width: 45rem) {
    flex-direction: row;
    margin-bottom: $l-size;
  }
}

.input-group__item {
  margin-bottom: $s-size;
  @media (min-width: 45rem) {
    margin: 0 $s-size 0 0;
  }
}
