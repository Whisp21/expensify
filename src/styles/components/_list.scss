.list-body {
  margin-bottom: $m-size;
  @media (min-width: 45rem) {
    margin-bottom: $l-size;
  }
}

.list-header {
  background: $off-white;
  border: 1px solid darken($off-white, 10%);
  color: $grey;
  display: flex;
  justify-content: space-between;
  padding: $s-size $m-size;
}

.list-item {
  border: 1px solid darken($off-white, 10%);
  border-top: none;
  color: $dark-grey;
  display: flex;
  flex-direction: column;
  padding: $m-size;
  text-decoration: none;
  transition: background .3s ease;
  &:hover {
    background: $off-white;
  }
  @media (min-width: 45rem) {
    align-items: center;
    flex-direction: row;
    justify-content: space-between;
    padding: $m-size;
  }
}

.list-item--message {
  align-items: center;
  color: $grey;
  justify-content: center;
  padding: $m-size;
  padding: $m-size;
  &:hover {
    background: none;
  }
}

.list-item__title {
  margin: 0;
  word-break: break-all;
}

.list-item__date {
  color: $grey;
  font-size: $font-size-small;
}

.list-item__amount {
  margin: $s-size 0 0 0;
  @media (min-width: 45rem) {
    margin: 0;
    padding-left: $s-size;
  }
}
