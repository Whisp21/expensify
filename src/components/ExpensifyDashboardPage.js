import React from "react";
import ConnectedExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";
import ExpenseSummary from "./ExpenseSummary"

const ExpensifyDashboardPage = () => (
  <div>
    <ExpenseSummary />
    <ExpenseListFilters />
    <ConnectedExpenseList />
  </div>
);

export default ExpensifyDashboardPage;
