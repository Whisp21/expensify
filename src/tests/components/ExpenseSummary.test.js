import React from "react";
import { shallow } from "enzyme";
import { ExpenseSummary } from "../../components/ExpenseSummary";

test("should render ExpenseSummary with one expense", () => {
  const wrapper = shallow(<ExpenseSummary expensesCount={1} expensesTotal={100}/>);
  expect(wrapper).toMatchSnapshot();
});

test("should render ExpenseSummary with multiple expenses", () => {
  const wrapper = shallow(<ExpenseSummary expensesCount={150} expensesTotal={2350} />);
  expect(wrapper).toMatchSnapshot();
});
