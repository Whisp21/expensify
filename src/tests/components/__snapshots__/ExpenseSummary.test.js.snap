// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      <span>
        150
      </span>
       
      expenses
       totalling 
      <span>
        R 
        23.50
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <[object Object]
        className="button add--button"
        to="/create"
      >
        Add Expense
      </[object Object]>
    </div>
  </div>
</div>
`;

exports[`should render ExpenseSummary with one expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        R 
        1.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <[object Object]
        className="button add--button"
        to="/create"
      >
        Add Expense
      </[object Object]>
    </div>
  </div>
</div>
`;
